transport/shadowsocks/salt_test.go

```diff
@@ -42,3 +42,57 @@ func BenchmarkRandomSaltGenerator(b *testing.B) {
 		}
 	})
 }
+
+// setRandomBitsToOne replaces any random bits in the output with 1.
+func setRandomBitsToOne(salter SaltGenerator, output []byte) error {
+	salt := make([]byte, len(output))
+	// OR together 128 salts.  The probability that any random bit is
+	// 0 for all 128 random salts is 2^-128, which is close enough to zero.
+	for i := 0; i < 128; i++ {
+		if err := salter.GetSalt(salt); err != nil {
+			return err
+		}
+		for i := range salt {
+			output[i] |= salt[i]
+		}
+	}
+	return nil
+}
+
+// Test that the prefix bytes are respected, and the remainder are random.
+func TestTypicalPrefix(t *testing.T) {
+	prefix := []byte("twelve bytes")
+	salter := NewPrefixSaltGenerator(prefix)
+
+	output := make([]byte, 32)
+	if err := setRandomBitsToOne(salter, output); err != nil {
+		t.Error(err)
+	}
+
+	for i := 0; i < 12; i++ {
+		if output[i] != prefix[i] {
+			t.Error("prefix mismatch")
+		}
+	}
+
+	for _, b := range output[12:] {
+		if b != 0xFF {
+			t.Error("unexpected zero bit")
+		}
+	}
+}
+
+// Test that all bytes are random when the prefix is nil
+func TestNilPrefix(t *testing.T) {
```

Comment:
Maybe add test case for empty array in this function as well?

```suggestion
func TestNilOrEmptyPrefix(t *testing.T) {
```

Reply:
We are passing `nil` in the test, not an empty `[]byte`